src/MeshKernel/CurvilinearGrid/CurvilinearGrid.cpp

```diff
@@ -603,20 +603,95 @@ CurvilinearGrid::ComputeDirectionalSmoothingFactors(NodeIndices const& gridpoint
     const auto maxHorizontalDelta = gridpoint.m_m > pointOnSmoothingLineIndices.m_m ? upperRightIndices.m_m - pointOnSmoothingLineIndices.m_m : pointOnSmoothingLineIndices.m_m - lowerLeftIndices.m_m;
-    const auto horizontalSmoothingFactor = maxHorizontalDelta == 0 ? 1.0 : (1.0 + std::cos(M_PI * static_cast<double>(horizontalDelta) / static_cast<double>(maxHorizontalDelta))) * 0.5;
+    const auto mSmoothingFactor = maxHorizontalDelta == 0 ? 1.0 : (1.0 + std::cos(M_PI * static_cast<double>(horizontalDelta) / static_cast<double>(maxHorizontalDelta))) * 0.5;
 
     // vertical smoothing factor
     const auto verticalDelta = gridpoint.m_n > pointOnSmoothingLineIndices.m_n ? gridpoint.m_n - pointOnSmoothingLineIndices.m_n : pointOnSmoothingLineIndices.m_n - gridpoint.m_n;
     const auto maxVerticalDelta = gridpoint.m_n > pointOnSmoothingLineIndices.m_n ? upperRightIndices.m_n - pointOnSmoothingLineIndices.m_n : pointOnSmoothingLineIndices.m_n - lowerLeftIndices.m_n;
-    const auto verticalSmoothingFactor = maxVerticalDelta == 0 ? 1.0 : (1.0 + std::cos(M_PI * static_cast<double>(verticalDelta) / static_cast<double>(maxVerticalDelta))) * 0.5;
+    const auto nSmoothingFactor = maxVerticalDelta == 0 ? 1.0 : (1.0 + std::cos(M_PI * static_cast<double>(verticalDelta) / static_cast<double>(maxVerticalDelta))) * 0.5;
 
     // mixed smoothing factor
-    const auto mixedSmoothingFactor = std::sqrt(verticalSmoothingFactor * horizontalSmoothingFactor);
+    const auto mixedSmoothingFactor = std::sqrt(nSmoothingFactor * mSmoothingFactor);
 
-    return {horizontalSmoothingFactor, verticalSmoothingFactor, mixedSmoothingFactor};
+    return {mSmoothingFactor, nSmoothingFactor, mixedSmoothingFactor};
 }
 
 CurvilinearGrid CurvilinearGrid::CloneCurvilinearGrid() const
 {
     return CurvilinearGrid(m_gridNodes, m_projection);
 }
+
+double CurvilinearGrid::ComputeAverageNodalDistance(NodeIndices const& index, GridLineDirection direction)
+{
+    if (index.m_m > m_gridNodes.size() || index.m_n > m_gridNodes[0].size())
+    {
+        throw std::invalid_argument("CurvilinearGrid::ComputeAverageNodalDistance: invalid index coordinates");
+    }
+
+    if (direction == GridLineDirection::MDirection)
+    {
+        double numEdges = 0.0;
+        double leftDistance = 0.0;
+        double rightDistance = 0.0;
+        if (index.m_m > 0 && m_gridNodes[index.m_m - 1][index.m_n].IsValid())
+        {
+            leftDistance = ComputeDistance(m_gridNodes[index.m_m][index.m_n], m_gridNodes[index.m_m - 1][index.m_n], m_projection);
+            numEdges += 1.0;
+        }
+        if (index.m_m + 1 < m_gridNodes.size() && m_gridNodes[index.m_m + 1][index.m_n].IsValid())
+        {
+            rightDistance = ComputeDistance(m_gridNodes[index.m_m][index.m_n], m_gridNodes[index.m_m + 1][index.m_n], m_projection);
+            numEdges += 1.0;
+        }
+        return (leftDistance + rightDistance) / numEdges;
```

Comment:
if neither of the branches was visited this results in a division by 0

Reply:
done